### ISSUE TITLE
Newsletter for members

### ISSUE THREAD

**Issue by DavidHernandez** [AsociacionDrupalES/asociaciondrupal#47]
Hola!

The Marketing group has been discussing how to improve the communication with the AED members and has decided to replace the mailing list as main communication method with a newsletter.

For now, we can send the newsletter to all the members of the AED (all registered users with a member role, even if is not the current role). But it will be great if we can select which roles we want to receive the newsletter. For example, to send the coupons for the DrupalCamp.

Also, the Marketing group has been discussing to open this Newsletter system to everybody, so in the future we might want to be able to send the mail to all the registered users.

And to follow the LOPD law, we need to add a control to allow the user to choose not to receive the newsletter. Maybe a checkbox on the profile?

And that's it for now. The Marketing group has a task to provide a design for the newsletter. If they fail to do so, or the newsletter is ready before that, we can go with a basic header using the AED logo.

Thanks,
David.

**Comment by DavidHernandez** [comment]
Attached a first design for the newsletter, provided by @eidoscom

![newsletter](https://cloud.githubusercontent.com/assets/877002/6226364/a5c5ecf8-b694-11e4-8f43-83d532c653b3.jpg)

**Comment by marcoscano** [comment]
I'm starting to evaluate the mailchimp for a basic newsletter functionality.
Question: Do we want all new users to be automatically included in the newsletter list or they need to explicitly opt in by checking the option on their profile page? (if they are automatically included, they can always opt out from the profile page or from the mailchimp footer links at any time)

**Comment by marcoscano** [comment]
On absence of any response I assume that yes users will be auto-subscribed :)
( In any case this can be changed later on. )

Craeted a [PR](https://github.com/AsociacionDrupalES/asociaciondrupal/pull/53) for the first mailchimp subscription functionality.

In order to test, an active account on mailchimp with at least one list is necessary. I have created one for this preliminary testing, ask me the credentials if you need them.

Steps to test:
1) Make sure you have a mailchimp account with at least one list created.
2) Define in your settings.php file the following variables:
$conf['mailchimp_api_key'] = 'xxx';
$conf['aed_mailchimp_subscription_mc_list_id'] = 'xxx'; // This is the list ID.
  (ask me on IRC for these credentials if you want to use the testing account I just created)
3) drush en aed_mailchimp_subscription --y
4) All existing users will have a new checkbox on their account edit page to subscribe to / unsubscribe from the newsletter (unchecked).
5) All new users will have this checkbox selected by default. (in order to test this you might want to enable the "test payment" method to register test users without paying :)
6) There is a new VBO view available at /admin/mailchimp-bulk-operations that allows mass subscribing / unsubscribing of users.

Remaining tasks:
1) Test the subscription process implemented until now
2) Transform the design into a mailchimp template (html)
3) Test the process of sending newsletters manually from mailchimp, once the template is OK
4) Evaluate if we want to send newsletters from drupal, using the module Mailchimp Campaings.

**Comment by marcoscano** [comment]
Side note: there is some minor css styling needed in the user form, but I was told not to bother with this now because a complete new theme is in-progress. Let me know if this needs to be corrected before the new theme gets done.

**Comment by DavidHernandez** [comment]
As we don't know when the new theme will be ready, my opinion is to fix the CSS that is required.

In my opinion, we should add the MailChimp Campaings module. So we don't have to give the access to the mailchimp account.

And there should be a block so anonymous visitors can join the newsletter.

Thanks for the good work!

**Comment by marcoscano** [comment]
Done some more work on this.

In my opinion the easiest way to manage the newsletters is to have 2 separate lists on mailchimp, one for members (socios) and another one for non-members (no socios).

I have updated the [PR](https://github.com/AsociacionDrupalES/asociaciondrupal/pull/53) and now we have a block on the sidebar with the signup form intended for the non member users (anonymous).

The campaigns module is already a dependency, so it will be available for anyone to create campaigns from the web (drupal), without the need for the editors to access the mailchimp account. It might be easier, though, to have our design implemented as a mailchimp template in the mailchimp side, this way the template could be imported automatically and modified by the person creating the campaign. In any case, it doesn't seem to be necessary, we can just paste the whole HTML code inside the drupal form when creating the campaign.

So the updated instructions for testing this are:

Steps to test:
1) Make sure you have a mailchimp account with at least one list created.
2) Define in your settings.php file the following variables:
$conf['mailchimp_api_key'] = 'xxx';
$conf['aed_mailchimp_subscription_mc_list_id'] = 'xxx'; // This is the list ID.
$conf['aed_mailchimp_subscription_mc_nm_list_id'] = 'xxx'; // This is the non-members list ID.
(ask me on IRC for these credentials if you want to use the testing account I just created)
3) drush en aed_mailchimp_subscription --y
4) All existing users will have a new checkbox on their account edit page to subscribe to / unsubscribe from the newsletter (unchecked).
5) All new users will have this checkbox selected by default. (in order to test this you might want to enable the "test payment" method to register test users without paying :)
6) There is a new VBO view available at /admin/mailchimp-bulk-operations that allows mass subscribing / unsubscribing of users.
7) There is a new block on the sidebar, visible only to anonymous users, with a signup form that allows visitors to subscribe to the list defined in the settings.php.

Remaining tasks:
1) Test the subscription processes implemented until now
2) Transform the design into a mailchimp template (html) [optional]
3) Test the process of sending newsletters manually from mailchimp, once the template is OK
4) Test the process of sending newsletters from drupal (/admin/config/services/mailchimp/campaigns)
5) Confirm the legal stuff (LOPD). For example, we probably need to have a checkbox for the anonymous visitors below the signup form, for TOS acceptance. But in this case we should check that the TOS text we link from there has the legal information updated.

**Comment by DavidHernandez** [comment]
Awesome, great work!

@capynet do we have a dev / test environment where we can check if everything works correctly?

**Comment by capynet** [comment]
Yes of course @DavidHernandez. dev.asociaciondrupal.es I deployed the code and synced db and files from production.

https://github.com/AsociacionDrupalES/asociaciondrupal/wiki/Contribuir-en-la-web-de-la-AED

**Comment by capynet** [comment]
@marcoscano what do you need to finish this issue?
Currently I synced dev with prod. Right now all code pushed on dev is deployed.

**Comment by marcoscano** [comment]
In my opinion the remaining tasks are not exclusively technical, in the sense we might need input from MKT to help on "business" decisions, for example:
 - How newsletter editors will create the newsletters (mailchimp site, plain html, modifying a template, etc)
 - The legal stuff for the anonymous signup (LOPD), etc.

So my suggestion is that the QA testing on this is done together with the missing definitions. Perhaps we could ask the MKT team to have a look on the testing instructions indicated above and discuss it on the next meeting to decide on how to manage the following steps.
